com.unity.shadergraph/Editor/GraphUI/DataModel/ShaderGraphModel.cs

```diff
@@ -276,5 +300,26 @@ public static bool ShouldElementBeVisibleToSearcher(ShaderGraphModel shaderGraph
                 return false;
             }
         }
+
+        public override IVariableNodeModel CreateVariableNode(IVariableDeclarationModel declarationModel,
+            Vector2 position, SerializableGUID guid = default, SpawnFlags spawnFlags = SpawnFlags.Default)
+        {
+            Action<VariableNodeModel> initCallback = variableNodeModel =>
+            {
+                if (declarationModel is GraphDataVariableDeclarationModel model && variableNodeModel is GraphDataVariableNodeModel graphDataVariable)
+                {
+                    variableNodeModel.VariableDeclarationModel = model;
+                    // Every time a variable node is added to the graph, add a node referencing that variable on the CLDS level
```

Comment:
should amend this comment to: "add a reference node pointing back to the variable/property that is wrapped by the VariableDeclarationModel, on the CLDS level"